python-package/lightgbm/basic.py

```diff
@@ -1853,6 +1853,115 @@ def free_network(self):
         self.network = False
         return self
 
+    def trees_to_dataframe(self):
+        """Parse the fitted model and return in an easy-to-read pandas DataFrame.
+
+        Returns
+        -------
+        Dataframe or list
+            Returns a pandas Dataframe if pandas is installed. Otherwise returns a list.
+        """
+        if not PANDAS_INSTALLED:
+            print('Pandas is not installed and so the result will be returned as a list')
+
+        if self.num_trees() == 0:
+            print('There are no trees in this Booster and thus nothing to return')
+            return
+
+        def _is_split_node(t):
+            return 'split_index' in t.keys()
+
+        def create_node_record(t, node_depth=1, tree_index=None,
+                               feature_names=None, parent_node=None):
+
+            def _get_node_index(t, tree_index):
+                tree_num = str(tree_index) + '-' if tree_index is not None else ''
+                node_type = 'S' if _is_split_node(t) else 'L'
+                node_num = str(
+                    t['split_index' if _is_split_node(t) else 'leaf_index'])
+                return tree_num + node_type + node_num
+
+            def _get_split_feature(t, feature_names):
+                if _is_split_node(t):
+                    if feature_names is not None:
+                        feature_name = feature_names[t['split_feature']]
+                    else:
+                        feature_name = t['split_feature']
+                else:
+                    feature_name = None
+                return feature_name
+
+            # Create the node record, and populate universal data members
+            node = {'tree_index': tree_index,
```

Comment:
I don't see `internal_value`, `internal_weight`, `leaf_value` and `leaf_weight` fields in this dict.

Reply:
I've added all of these. Note `internal_weight` and `leaf_weight` both map to `weight` in the output which I assume is the sum of the hessians.